src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,192 @@
+                if (numberOfArcherBarbarians < 0)
+                {
+                    numberOfArcherBarbarians = 0;
+                }
+                hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+                if (hordeTotal > maxSize)
+                {
+                    numberOfChiefBarbarians = hordeTotal - maxSize;
+
+                    //For error handling and correct hordeTotal
+                    if (numberOfChiefBarbarians < 0)
+                    {
+                        numberOfChiefBarbarians = 0;
+                    }
+                }
+            }
+        }
+
+        int x = colony.getCenter().getX();
+        int y = colony.getCenter().getY();
+        int z = colony.getCenter().getZ();
+
+        //Make sure world isn't null
+        if (raidingWorld == null)
+        {
+            return;
+        }
+        switch (raidingWorld.rand.nextInt(7))
+        {
+            case 0:
+                x += Configurations.workingRangeTownHall + 20;
+                break;
+            case 1:
+                x -= Configurations.workingRangeTownHall + 20;
+                break;
+            case 2:
+                z += Configurations.workingRangeTownHall + 20;
+                break;
+            case 3:
+                z -= Configurations.workingRangeTownHall + 20;
+                break;
+            case 4:
+                x += Configurations.workingRangeTownHall + 20;
+                z += Configurations.workingRangeTownHall + 20;
+                break;
+            case 5:
+                x += Configurations.workingRangeTownHall + 20;
+                z -= Configurations.workingRangeTownHall + 20;
+                break;
+            case 6:
+                x -= Configurations.workingRangeTownHall + 20;
+                z += Configurations.workingRangeTownHall + 20;
+                break;
+            case 7:
+                x -= Configurations.workingRangeTownHall + 20;
+                z -= Configurations.workingRangeTownHall + 20;
+                break;
+            default:
+                x += Configurations.workingRangeTownHall + 20;
+                break;
+        }
+
+        y = raidingWorld.getTopSolidOrLiquidBlock(new BlockPos.MutableBlockPos(x, y, z)).getY(); //Make sure mob spawns on surface.
+
+        spawn(BarbarianUtils.barbarian, numberOfBarbarians, x, y, z, raidingWorld);
+        spawn(BarbarianUtils.archer, numberOfArcherBarbarians, x, y, z, raidingWorld);
+        spawn(BarbarianUtils.chief, numberOfChiefBarbarians, x, y, z, raidingWorld);
+    }
+
+    /**
+     * Spawns EntityToSPawn at the X, Y, Z
+     *
+     * @param entityToSpawn
+     * @param numberOfSpawns
+     * @param x
+     * @param y
+     * @param z
+     * @param world
+     */
+
+    public static void spawn(final ResourceLocation entityToSpawn, int numberOfSpawns, int x, int y, int z, final World world)
+    {
+        IntStream.range(0, numberOfSpawns).forEach($ ->
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this lambda expression number of lines from 27 to at most 20. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)